CHANGELOG.md

```diff
@@ -4,6 +4,234 @@ Here you can find upgrade changes in between releases and upgrade instructions.
 
 ## UNRELEASED
 
+## 2.0
+
+### 2.0.0
+
+#### Highlights
+
+- **TODO**
+
+  TODO
+
+#### Security critical upgrade notice
+
+This is only relevant if you have configured `...networkPolicy.egress` for
+`hub`, `proxy.chp`, `proxy.traefik`, or `singleuser`.
+
+If you have configured `...networkPolicy.egress` for `hub`, `proxy.chp`,
+`proxy.traefik` or `singleuser` to restrict the permissions to establish
+outbound network connections, this upgrade is likely to escalate those
+permissions unless you revise your configuration. The new configuration
+`...networkPolicy.egressAllowRules` are by default granting most of the egress
+permissions previously granted by default via the `...networkPolicy.egress`
+configuration, and `...networkPolicy.egress` are now by default not providing
+any permissions.
+
+If you for example had overridden the previously very permissive default value
+of `singleuser.networkPolicy.egress` to be a bit less permissive, you should if
+you are unsure disable all `singleuser.networkPolicy.egressAllowRules` like this
+to not risk escalating the permissions.
+
+```yaml
+singleuser:
+  networkPolicy:
+    egressAllowRules:
+      cloudMetadataServer: false
+      dnsPortsPrivateIPs: false
+      nonPrivateIPs: false
+      privateIPs: false
+```
+
+For more details, see the documentation on [Kubernetes Network
+Policies](https://zero-to-jupyterhub.readthedocs.io/en/latest/administrator/security.html#kubernetes-network-policies)
+and the configuration reference entries under
+[`...networkPolicy.egress`](https://zero-to-jupyterhub.readthedocs.io/en/latest/resources/reference.html#hub-networkpolicy-egress)
+and
+[`...networkPolicy.egressAllowRules`](https://zero-to-jupyterhub.readthedocs.io/en/latest/resources/reference.html#hub-networkpolicy-egressallowrules).
+
+#### Upgrade instructions
+
+1. Read and adjust your configuration to the security critical upgrade notice
+   above if it applies to your deployment.
+2. Prepare a user image with a modern version of JupyterHub installed. Install
+   `jupyterhub>=2.3.1` via pip or `jupyterhub-base>=2.3.1` via conda/mamba in
+   the image. Configure to use this new image, which is typically done via
+   `singleuser.image` of as part of `singleuser.profileList`.
+3. Read and adjust your configuration to the breaking changes below.
+
+#### Breaking changes
+
+- The following PRs have breaking labels since 1.1.3 release (1.1.4 and 1.2.0 are backport prs)
+
+  - use image for singleuser.cmd by default - https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2449
```

Comment:
```suggestion
  - use image for singleuser.cmd by default - https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2449
    - Z2JH defaults to using the default entrypoint and command from the singleuser container image. If your container image sets a command such as `jupyterhub-singleuser` and you want to add custom arguments you must include the command.
```
Note `default_url` specifically mentioned in https://github.com/jupyterhub/kubespawner/issues/493 shouldn't matter since JupyterHub 2.0.0 made all arguments managed by JupyterHub into environment variables https://github.com/jupyterhub/jupyterhub/pull/3381/files

Reply:
Maybe give the exact config as an example:

```suggestion
  - use image for `singleuser.cmd` by default - https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2449
    - Z2JH defaults to using the default entrypoint and command from the singleuser container image. If you want to add custom arguments to the command, you must specify the full command and any arguments in `singleuser.cmd`, for example:
      ```yaml
      singleuser:
        cmd:
          - jupyterhub-singleuser
          - "--colaborative"
          - "--debug"
      ```
```

Reply:
Added to `upgrade-1-to-2.md`